### PR TITLE
Multiple model export issue is fixed

### PR DESCRIPTION
**Pull request by IevhenIkonnykov**
[Fix] an issue for a SiteReferenceDataLibrary HashSet
[Add] a little bit more comprehensible error messages from the server and some necessary checks

### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/CDP4-WebServices-Community-Edition/pulls) open
- [x] I have verified that I am following the CDP4-SDK [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/CDP4-WebServices-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
The issue was that SiteReferenceDataLibrary HashSet relied on default equality implementation which is link comparison, but the code produces SRDLs dtos that are equal, but because they have different links they were added and zip failed for duplicates.

### CODE REVIEW

**Comment by IevhenIkonnykov** [comment]
Closes #54

File: CDP4WebServices.API/Services/BusinessLogic/Implementation/EngineeringModelZipExportService.cs [hunk 1]
```diff
@@ -1045,5 +1051,28 @@ private void WriteHeaderToZipFile(ExchangeFileHeader echExchangeFileHeader, ZipF
             credentials.IsParticipant = currentParticipantFlag;
             this.PermissionService.Credentials = credentials;
         }
+
+        /// <summary>
+        /// An implementation of <see cref="IEqualityComparer{Thing}"/> for <see cref="CDP4Common.DTO.Thing"/> to be used in <see cref="HashSet{Thing}"/>.
+        /// </summary>
+        private class ThingComparer : IEqualityComparer<Thing>
```
Comment (alexatrhea) [inline CDP4WebServices.API/Services/BusinessLogic/Implementation/EngineeringModelZipExportService.cs]: Please have a good look whether this is already available somewhere? And if not for sure move it to own file and maybe call it `ThingIidComparer`.

Comment (samatrhea) [reply CDP4WebServices.API/Services/BusinessLogic/Implementation/EngineeringModelZipExportService.cs]: I agree with @alexatrhea that the name should be `ThingIidComparer`. I also think that this belongs in CDP4Common in the SDK as a public class.

A potential place (namespace) to put it in could be `CDP4Common.Comparers`

**Comment by IevhenIkonnykov** [comment]
The latest change to domainOfExpertise export closes #55 . I used CRANFIELD data, created a model based on the existing one and then exported these two model. After that I tried to seed exported data and it succeeded. It turned out that existing logic produced id duplicates of domainOfExpretise in SiteDirectory which did not cause problems during an export process, but led to a failure during a seed process. I would appreciate if someone could verify it by using a different model.
Once changes in CDP4-SDK (https://github.com/RHEAGROUP/CDP4-SDK-Community-Edition/pull/84) are published I will update dependecies and exclude the private comparer by utilising one from SDK.

**Comment by IevhenIkonnykov** [comment]
The instruction is attached with Postman collection export file.
[MultipleModelVerification.txt](https://github.com/RHEAGROUP/CDP4-WebServices-Community-Edition/files/3925789/MultipleModelVerification.txt)

[CDP4WebServices.postman_collection.txt](https://github.com/RHEAGROUP/CDP4-WebServices-Community-Edition/files/3925797/CDP4WebServices.postman_collection.txt)

Please rename the extension of  CDP4WebServices.postman_collection.txt from txt to json just in case. GitHub does not allow to upload json files.

**Pull request merged**